app/controllers/application_controller.rb

```diff
@@ -47,6 +48,14 @@ def user_type
     end
   end
 
+  def jobseeker_root_path
+    if current_jobseeker.job_applications.any?
+      jobseekers_job_applications_path
+    else
+      jobseekers_saved_jobs_path
+    end
+  end
```

Comment:
A small issue here is that the entry you've removed from `routes.rb` does a little more than just this.  You'd want to:

1. also define `jobseeker_root_url`
2. pass any args through to the more specific methods in case people are passing stuff in

Reply:
```suggestion
  def jobseeker_root_path(*args)
    if current_jobseeker.job_applications.any?
      jobseekers_job_applications_path(*args)
    else
      super(*args)
    end
  end

  def jobseeker_root_url(*args)
    if current_jobseeker.job_applications.any?
      jobseekers_job_applications_url(*args)
    else
      super(*args)
    end
  end
```